packages/mcfunction/src/parser/macro.ts

```diff
@@ -0,0 +1,122 @@
+import * as core from '@spyglassmc/core'
+import { localeQuote, localize } from '@spyglassmc/locales'
+import type {
+	MacroGapNode,
+	MacroKeyNode,
+	MacroNode,
+	MacroSignNode,
+	MacroVariableNode,
+} from '../node/index.js'
+
+/**
+ * Parse a macro line.
+ */
+export function macro(): core.Parser<MacroNode> {
+	return (
+		src: core.Source,
+		ctx: core.ParserContext,
+	): core.Result<MacroNode> => {
+		const children:
+			(MacroGapNode | MacroVariableNode | MacroKeyNode | MacroSignNode)[] =
+				[]
+
+		// Skip the starting '$'
+		children.push({
+			type: 'mcfunction:macro/sign',
+			range: core.Range.create(src.cursor, src.cursor + 1),
+			value: '$',
+			path: ['$'],
+		})
+		src.skip()
+
+		let start = src.cursor
+		let hasMacroKeys = false
+
+		while (src.canReadInLine()) {
+			src.skipUntilOrEnd(core.LF, core.CR, '$')
+			if (src.peek(2) === '$(') {
+				// Add the gap before this macro key
+				const gap = src.sliceToCursor(start)
+				if (gap.length > 0) {
+					children.push({
+						type: 'mcfunction:macro/gap',
```

Comment:
So `macro/gap` is everything between macro arguments?

Reply:
Yes

Reply:
Gap makes me think of whitespaces. Would something like `macro/raw_text` be a more fitting name?

Reply:
Yes

Reply:
I had "other" originally but went with Misode's suggestion of "gap". Not personally a fan of "raw_text" but maybe another alternative?

Reply:
maybe `macro/text_body`?